### PR TITLE
Revert NPM non-critical functional changes in 7.34

### PR DESCRIPTION
**Pull request by brycekahle**
### What does this PR do?

Reverts all NPM functional changes from 7.34 that aren't considered critical fixes.

### Motivation

Memory issues

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by ISauve — APPROVED** [review]

File: .circleci/config.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ experimental:
 templates:
   job_template: &job_template
     docker:
-      - image: datadog/datadog-agent-runner-circle:go11612
```
Comment (mx-psi) [inline .circleci/config.yml:18]: This also affects all other components, are we sure it's a good idea to change it so late in the release process?

Comment (olivielpeau) [reply .circleci/config.yml:18]: I agree that we don't want to include the revert of the Go upgrade from 1.16.7 to 1.16.12 (https://github.com/DataDog/datadog-agent/pull/10404) as part of this PR, unless it's directly related to the NPM regression.

1.16.12 includes security fixes that we want to keep in 7.34

**Review by olivielpeau — CHANGES_REQUESTED** [review]

**Comment by brycekahle** [comment]
Updated to remove the Go revert

**Review by urseberry — APPROVED** [review]

**Review by juliogreff — APPROVED** [review]

**Review by olivielpeau — APPROVED** [review]
I haven't reviewed the full changeset, but the Go revert is gone, so looks good to me.

`kitchen_test_system_probe_linux_x64` is failing on centos and rhel (https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/117840540), since the job also fails on 7.34.0-rc.10 I'll ignore the failure and merge, but would be worth confirmation from you that the specific tests failures are expected @brycekahle

**Review by mx-psi — APPROVED** [review]

**Pull request merged**